src/pin.rs

```diff
@@ -0,0 +1,138 @@
+
+use crate::common::Generic;
+use crate::error::MockError;
+
+use embedded_hal::digital::v2::{OutputPin, InputPin};
+
+/// MockPin transaction
+#[derive(Debug)]
+pub struct Transaction {
+    kind: TransactionKind,
+    err: Option<MockError>,
```

Comment:
This is for "faking" an error during a transaction, right?

Reply:
yep, added some documentation and a test. i guess rustdocs would be a good addition too.

Reply:
Thanks! Shouldn't docs be sufficiently covered by https://docs.rs/?

Reply:
sorry i meant doctests, which are now there too.